src/main/java/de/retest/recheck/ui/image/Screenshot.java

```diff
@@ -32,14 +32,13 @@ public String getFileExtension() {
 	@XmlElement
 	private final String persistenceId;
 
-	/**
-	 * This should be final, too. When the corresponding setter is invoked, the persistence ID is <em>not</em> updated.
-	 * However, as we take care of this in our surrounding implementation, this shouldn't be an issue. Nonetheless, one
-	 * should note that this class is broken since it can be in a corrupt state.
-	 */
 	@XmlTransient
 	private byte[] binaryData;
 
+	// TODO After recheck 2.0 remove transient and ensureSha
```

Comment:
Why persisting the checkum? The very unlikely case that the image data is corrupted isn't really a problem.

Reply:
Good point. And the way it's now implemented, this isn't an issue anymore.